### PR TITLE
chore: set minimum compilation target to ES2021 (>= Node.js 16.4.x)

### PR DESCRIPTION
**Pull request by jkomyno**
This PR normalises the minimum compilation target to `ES2021`, and reverts https://github.com/prisma/prisma/pull/17133/.

As per [this table](https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping), `ES2021` is compatible with Node.js 16.4.x.
Prisma's minimum supported Node.js version is 16.13.x ([docs](https://www.prisma.io/docs/orm/reference/system-requirements)).

### CODE REVIEW

File: packages/cli/src/__tests__/fixtures/broken-example-project/tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "target": "ES2020" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */,
+    "target": "ES2021" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */,
```
Comment (Druue) [inline packages/cli/src/__tests__/fixtures/broken-example-project/tsconfig.json:3]: Does having these target comments still make sense to have? especially now that we're specifying a version not mentioned.
If we want to simply update it

```suggestion
    "target": "ES2021" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', 'ES2021', 'ES2022', or 'ESNEXT'. */,
```

File: packages/cli/src/__tests__/fixtures/example-project/tsconfig.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "target": "ES2020" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */,
+    "target": "ES2021" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */,
```
Comment (Druue) [inline packages/cli/src/__tests__/fixtures/example-project/tsconfig.json:3]:
```suggestion
    "target": "ES2021" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', 'ES2021', 'ES2022', or 'ESNEXT'. */,
```

File: packages/cli/src/__tests__/fixtures/init/tsconfig.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "target": "ES2020" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */,
+    "target": "ES2021" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */,
```
Comment (Druue) [inline packages/cli/src/__tests__/fixtures/init/tsconfig.json:3]:
```suggestion
    "target": "ES2021" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', 'ES2021', 'ES2022', or 'ESNEXT'. */,
```

File: packages/client/fixtures/blog/tsconfig.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
     "strict": true,
-    "target": "ES2020",
+    "target": "ES2021",
```
Comment (Druue) [inline packages/client/fixtures/blog/tsconfig.json:4]: Given that we don't include them consistently and that they are a changing list, I'm inclined to say that we just omit the comments in the other `tsconfig` files, thoughts?

File: packages/migrate/src/__tests__/fixtures/introspect/tsconfig.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "target": "ES2020" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */,
+    "target": "ES2021" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */,
```
Comment (Druue) [inline packages/migrate/src/__tests__/fixtures/introspect/tsconfig.json:3]:
```suggestion
    "target": "ES2021" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', 'ES2021', 'ES2022', or 'ESNEXT'. */,
```

**Review by Druue — APPROVED** [review]
The above comments are non-blocking however, this looks good 👍

File: packages/cli/src/__tests__/fixtures/broken-example-project/tsconfig.json (hunk 1 above)
Comment (jkomyno) [reply packages/cli/src/__tests__/fixtures/broken-example-project/tsconfig.json:3]: I think we can completely get rid of them

**Pull request merged**